runtime/metricsview/ast.go

```diff
@@ -640,6 +709,72 @@ func (a *AST) buildBaseSelect(alias string, comparison bool) (*SelectNode, error
 
 	a.addTimeRange(n, tr)
 
+	if comparison && a.query.inlineBaseSelect {
+		// Add the dimensions values as a "<dim> IN (<vals...>)" expression in the outer query's WHERE clause.
+		for dim, values := range a.query.inlineDims {
+			var inExpr *Expression
+			if len(a.query.inlineDims) == 0 {
+				inExpr = &Expression{
+					Value: false,
+				}
+			} else {
+				// if any dim value is nil add condition with eq operator with nil value
+				var vals []any
+				foundNil := false
+				for _, k := range values {
+					if k == nil {
+						foundNil = true
+					} else {
+						vals = append(vals, k)
+					}
+				}
+				inExpr = &Expression{
+					Condition: &Condition{
+						Operator: OperatorIn,
+						Expressions: []*Expression{
+							{Name: dim.(string)},
```

Comment:
If `dim` comes from an OLAP query, I don't think you can be sure it's a string. As mentioned above, we support e.g. struct/map dims.

Reply:
using query dim name now directly from the query